dags/vds_pull_vdsdata.py

```diff
@@ -0,0 +1,153 @@
+import os
+import sys
+from airflow import DAG
+from datetime import datetime, timedelta
+from airflow.operators.python import PythonOperator
+from airflow.providers.postgres.hooks.postgres import PostgresHook
+from airflow.providers.postgres.operators.postgres import PostgresOperator
+from airflow.models import Variable
+from airflow.utils.task_group import TaskGroup
+from functools import partial
+
+dag_name = 'vds_pull_vdsdata'
+
+#CONNECT TO ITS_CENTRAL
+itsc_bot = PostgresHook('itsc_postgres')
+
+#CONNECT TO BIGDATA
+vds_bot = PostgresHook('vds_bot')
+
+# Get DAG Owner
+#dag_owners = Variable.get('dag_owners', deserialize_json=True)
+#names = dag_owners.get(dag_name, ['Unknown']) #find dag owners w/default = Unknown    
+names = ['gabe']
+
+#op_kwargs:
+conns = {'rds_conn': vds_bot, 'itsc_conn': itsc_bot}
+start_date = {'start_date': '{{ ds }}'}
+
+try:
+    repo_path = os.path.abspath(os.path.dirname(os.path.dirname(os.path.realpath(__file__))))
+    sys.path.insert(0,os.path.join(repo_path,'volumes/vds/py'))
+    from vds_functions import task_fail_slack_alert, pull_raw_vdsdata, pull_detector_inventory, pull_entity_locations
+except:
+    raise ImportError("Cannot import functions from volumes/vds/py/vds_functions.py.")
+   
+default_args = {
+    'owner': ','.join(names),
+    'depends_on_past': False,
+    'start_date': datetime(2023, 5, 1),
+    'email_on_failure': False,
+    'email_on_retry': False,
+    'retries': 5,
+    'retry_delay': timedelta(minutes=5),
+    'retry_exponential_backoff': True, #Allow for progressive longer waits between retries
+    'on_failure_callback': partial(task_fail_slack_alert, dag_name = dag_name, owners = names),
+    'catchup': True,
+}
+
+with DAG(dag_name,
+         default_args=default_args,
+         max_active_runs=5,
+         schedule_interval='0 4 * * *') as dag: #daily at 4am
+
+    #this task group deletes any existing data from RDS vds.raw_vdsdata and then pulls and inserts from ITSC
+    with TaskGroup(group_id='pull_vdsdata') as vdsdata:
+        #deletes data from vds.raw_vdsdata
+        delete_raw_vdsdata_task = PostgresOperator(
+            sql="""DELETE FROM vds.raw_vdsdata
+                    WHERE
+                    datetime_15min >= '{{ds}} 00:00:00'::timestamp
+                    AND datetime_15min < '{{ds}} 00:00:00'::timestamp + INTERVAL '1 DAY'""",
+            task_id='delete_vdsdata',
+            dag=dag,
```

Comment:
no need to pass the `dag` to these tasks as long as they are already defined within the DAG context